zenoh/src/api/session.rs

```diff
@@ -2162,12 +2171,37 @@ impl SessionInner {
         #[cfg(feature = "unstable")] source_info: SourceInfo,
         attachment: Option<ZBytes>,
     ) -> ZResult<()> {
+        const NO_REMOTE_FLAG: u64 = 0b01;
```

Comment:
Let's refactor `cache` to be an object with methods like `set_no_remotes()` etc to make code cleaner and more readable.

Reply:
The issue is that `resolve_put` is shared with `Session::put` workflow, in which there is no cache. Because I'm not sure that `resolve_put` is inlined, I've written the code in the "most efficient" way using a dummy `cached` value . But using a `PublisherCache` type when there is no publisher is quite odd...

Reply:
I will try to restructure the code